flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -676,7 +726,9 @@ private void returnExcessMemorySegments() {
     }
 
     private boolean hasExcessBuffers() {
-        return numberOfRequestedMemorySegments > currentPoolSize;
+        return numberOfRequestedOverdraftMemorySegments > 0
+                || numberOfRequestedMemorySegments + numberOfRequestedOverdraftMemorySegments
+                        > currentPoolSize;
```

Comment:
I think you overcomplicated the condition I believe this is an easier equivalent:
```
private boolean hasExcessBuffers() {
        return numberOfRequestedOverdraftMemorySegments > 0
                || numberOfRequestedMemorySegments > currentPoolSize;
    }
```